primitives/src/transaction.rs

```diff
@@ -291,6 +297,30 @@ fn hash_transaction(tx: &Transaction, uses_segwit_serialization: bool) -> sha256
     sha256d::Hash::from_engine(enc)
 }
 
+encoding::encoder_newtype! {
+    /// The encoder for the [`Transaction`] type.
+    pub struct TransactionEncoder<'e>(
+        Encoder4<ArrayEncoder<4>, LockTimeEncoder, SliceEncoder<'e, TxIn>, SliceEncoder<'e, TxOut>>
+    );
+}
+
+impl Encodable for Transaction {
+    type Encoder<'e>
+        =
+        Encoder4<ArrayEncoder<4>, LockTimeEncoder, SliceEncoder<'e, TxIn>, SliceEncoder<'e, TxOut>>
+    where
```

Comment:
nit: the `where Self: 'e` isn't required here, looks like the compiler figures it out. But I don't know if best to be explicit in these scenarios.